drivers/display/display_co5300.c

```diff
@@ -104,27 +100,34 @@ static int co5300_write(const struct device *dev,
 	const struct co5300_config *config = dev->config;
 	struct co5300_data *data = dev->data;
 	int ret;
-	uint16_t start_xpos;
-	uint16_t end_xpos;
-	uint16_t start_ypos;
-	uint16_t end_ypos;
+	uint16_t start_pos;
+	uint16_t end_pos;
 	const uint8_t *framebuffer_addr;
 	uint8_t cmd_params[4];
 	struct mipi_dsi_msg msg = {0};
 	uint32_t total_bytes_sent = 0U;
 	uint32_t framebuffer_size = 0U;
 	int bytes_written = 0;
 
+	if (((desc->width % 2) != 0U) || ((desc->height % 2) != 0U)) {
+		LOG_ERR("The height/width of the update area must be divisible by 2");
+		return -ENOTSUP;
+	}
+
 	LOG_DBG("WRITE:: W=%d, H=%d @%d,%d", desc->width, desc->height, x, y);
 
-	/* Set column address of target area */
+	/*
+	 * Set column address of target area. The circular panel actually starts
+	 * to show from row 6, row 0~5 are cut off physically. The actual display
+	 * area is row 6~472 and line 0~466. So adjust coordinates accordingly.
+	 */
 	/* First two bytes are starting X coordinate */
-	start_xpos = x;
```

Comment:
Should we update the width and height of the panel in the shield overlay instead of adding this runtime calculations?